manager/api/rest-impl/src/main/java/io/apiman/manager/api/rest/impl/util/SwaggerWsdlHelper.java

```diff
@@ -150,23 +150,25 @@ private static String jsonNodeToString(JsonNode jsonNode, ApiVersionBean apiVers
      */
     public static String updateLocationEndpointInWsdl(InputStream wsdlStream, URL managedEndpoint, ApiVersionBean apiVersion, IStorage storage) throws StorageException {
         Document document = readWsdlInputStream(wsdlStream);
-        NodeList nodeList = document.getDocumentElement().getElementsByTagName(SOAP_ADDRESS);
-
         Boolean updateStorage = false;
-        for (int i = 0; i < nodeList.getLength(); i++) {
-            Node node = nodeList.item(i);
-            String location = node.getAttributes().getNamedItem(LOCATION).getNodeValue();
-            String endpoint = managedEndpoint.toString();
-            if (!location.equals(endpoint)) {
-                node.getAttributes().getNamedItem(LOCATION).setNodeValue(managedEndpoint.toString());
-                updateStorage = true;
+        for(int i = 0; i < SOAP_ADDRESS.length;i++) {
+            NodeList nodeList = document.getDocumentElement().getElementsByTagName(SOAP_ADDRESS[i]);

```

Comment:
So I'm inclined to just merge this if it works for you guys.  And I didn't write this code originally and am not super familiar with what it's trying to accomplish.  However, this strikes me as being the wrong way to do this.  It seems that we're trying to find a specific element in a qualified/namespaced XML document using the non-qualified/non-namespaced XML API in Java.  This approach possibly indicates that the author of this code maybe didn't understand the difference?  Searching for a qualified XML element using `getElementsByTagName` is not going to reliably work (hence this change to include `soap12:address` as another option).  Instead, I think we should calling `getElementsByTagNameNS` and using a `QName` instead of a `String` to find the elements.  Here is what I think the code should be:

https://gist.github.com/EricWittmann/016d01846c181aa19d19f15a6be336ab

This only handles SOAP 1.2, but I don't think SOAP 1.1 is relevant to this...so it's likely OK.

@volkflo @tevosouza @bekihm  what do you guys think?

Reply:
@EricWittmann Yes you are right, I am not very familiar with the Java XML possibilities and at the time I wrote this I just googled a bit and used an approach that worked ;)
But I am totally fine if we change this and follow best practice :)

But we have WSDL like this: https://gist.github.com/volkflo/6b6c202af7a9573432d95d86964a15d9
And I think then they will not work with your `SOAP_NAMESPACE`?

Reply:
Yeah you guys are both right - WSDL files can have SOAP 1.0/1.1 **or** they can have SOAP 1.2 bindings.  I must have been looking at the wrong schema for the older version of SOAP.

In any case, we will need to search for the `address` element in both namespaces.  Here's what I'll do - I will merge this PR and then I will submit a separate PR that fixes this.